temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationTest.java

```diff
@@ -0,0 +1,134 @@
+import org.junit.Test;
+
+public class ChildWorkflowContinueAsNewCancellationTest {
+  private static AtomicInteger count;
+
+  @Rule
+  public SDKTestWorkflowRule testWorkflowRule =
+      SDKTestWorkflowRule.newBuilder()
+          .setWorkflowTypes(TestParentWorkflowImpl.class, TestChildWorkflowImpl.class)
+          .setActivityImplementations(new TestActivitiesImpl())
+          .build();
+
+  @Test
+  public void testChildWorkflowContinueAsNewCancellationTest() throws InterruptedException {
+    count = new AtomicInteger(3);
+    WorkflowStub parentWorkflow =
+        testWorkflowRule.newUntypedWorkflowStub("TestWorkflowCancellationType");
+    WorkflowExecution parentExecution =
+        parentWorkflow.start(ChildWorkflowCancellationType.WAIT_CANCELLATION_COMPLETED);
+    Thread.sleep(500);
+    parentWorkflow.cancel();
+    try {
+      parentWorkflow.getResult(String.class);
+      Assert.fail("unreachable");
+    } catch (WorkflowFailedException e) {
+      assertTrue(e.getCause() instanceof CanceledFailure);
+    }
+    testWorkflowRule.assertHistoryEvent(
+        parentExecution, EventType.EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_CANCELED);
+  }
+
+  @Test
+  public void testImposterWorkflowContinueAsNewCancellationTest() throws InterruptedException {
+    count = new AtomicInteger(3);
+    // Parent-child workflow
+    WorkflowStub parentWorkflow =
+        testWorkflowRule.newUntypedWorkflowStub("TestWorkflowCancellationType");
+    WorkflowExecution parentExecution =
+        parentWorkflow.start(ChildWorkflowCancellationType.WAIT_CANCELLATION_COMPLETED);
+    // Imposter workflow
+    WorkflowStub imposterChild = testWorkflowRule.newUntypedWorkflowStub("NoArgsWorkflow");
+    WorkflowExecution execution = imposterChild.start();
+    Thread.sleep(500);
+    parentWorkflow.cancel();
+    try {
+      parentWorkflow.getResult(String.class);
+      Assert.fail("unreachable");
+    } catch (WorkflowFailedException e) {
+      assertTrue(e.getCause() instanceof CanceledFailure);
+    }
+    // Assert parent behavior
+    // TODO: assert runID not set on this event and child workflow only boolean set to true
+    testWorkflowRule.assertHistoryEvent(
+        parentExecution, EventType.EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED);
+    testWorkflowRule.assertHistoryEvent(
+        parentExecution, EventType.EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_CANCELED);
+    // Assert imposter workflow behavior
+    testWorkflowRule.assertNoHistoryEvent(
+        execution, EventType.EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED);
+  }
+
+  public static class TestParentWorkflowImpl implements TestWorkflowCancellationType {
+    @Override
+    public void execute(ChildWorkflowCancellationType cancellationType) {
+      NoArgsWorkflow child =
+          Workflow.newChildWorkflowStub(
+              NoArgsWorkflow.class,
+              ChildWorkflowOptions.newBuilder().setCancellationType(cancellationType).build());
+      child.execute();
```

Comment:
Based on issue writeup I think this WF needs to issue a cancel (good to test signal too) to its child. Worth clarifying with Wenquan